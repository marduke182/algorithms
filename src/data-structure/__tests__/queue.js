import Queue from '../queue';

let consoleSpy;
let queue;

beforeEach(async () => {
  consoleSpy = jest.spyOn(console, 'log').mockImplementation(() => {});
  queue = new Queue(10);
});

test('should enqueue', async () => {
  queue.enqueue(1);
  queue.enqueue(2);
  queue.enqueue(3);
  queue.enqueue(4);

  queue.print();
  expect(consoleSpy).toBeCalledWith('1 2 3 4');
});


test('should dequeue', async () => {
  queue.enqueue(1);
  queue.enqueue(2);
  queue.enqueue(3);
  queue.enqueue(4);
  const value = queue.dequeue();

  queue.print();

  expect(consoleSpy).toBeCalledWith('2 3 4');
  expect(value).toEqual(1);
});

test('should peek the value', async () => {
  queue.enqueue(1);
  queue.enqueue(2);
  queue.enqueue(3);
  queue.enqueue(4);
  const value = queue.peek();

  queue.print();

  expect(consoleSpy).toBeCalledWith('1 2 3 4');
  expect(value).toEqual(1)
});

test('should throw if queue is full', async () => {
  queue = new Queue(1);
  queue.enqueue(1);

  expect(() => queue.enqueue(2)).toThrowErrorMatchingSnapshot();
});