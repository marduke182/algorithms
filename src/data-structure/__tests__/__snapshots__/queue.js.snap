// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should throw if queue is full 1`] = `"Queue is full."`;
